### ISSUE TITLE
UI not opening after update to 5.3.0-20200310205642+4f2689009b

### ISSUE THREAD

**Issue by carroarmato0** [keybase/client#23018]
I updated Keybase through my package manager on Ubuntu 19.10, but now the UI seems to be crashing immediately when starting up.

I do have a crash report from the crash detection system on Ubuntu 
[_opt_keybase_Keybase.1000.crash.txt](https://github.com/keybase/client/files/4316998/_opt_keybase_Keybase.1000.crash.txt)

**Comment by majk-p** [comment]
Same for me on Ubuntu 18.04:
```
$ /opt/keybase/Keybase                                                                                                                                
Version: 5.3.0-20200310205642+4f2689009b
(electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
(electron) 'setAutoHideMenuBar function' is deprecated and will be removed. Please use 'autoHideMenuBar property' instead.
[1]    8373 segmentation fault (core dumped)  /opt/keybase/Keybase
```

**Comment by heronhaye** [comment]
Thanks, we're looking into it.

**Comment by carroarmato0** [comment]
> Thanks, we're looking into it.

If you need any info from me that could help with troubleshooting, let me know

**Comment by heronhaye** [comment]
@carroarmato0 What's your kernel version? `uname -a`

**Comment by carroarmato0** [comment]
> @carroarmato0 What's your kernel version? `uname -a`

Linux neon-flower 5.3.0-40-generic #32-Ubuntu SMP Fri Jan 31 20:24:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by heronhaye** [comment]
@majk-p how about you?

**Comment by majk-p** [comment]
The same:
`Linux HOME 5.3.0-40-generic #32~18.04.1-Ubuntu SMP Mon Feb 3 14:05:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by heronhaye** [comment]
I've tried both 18.04 on a 4.15 kernel and 19.10 on 5.3 and I can't reproduce the issue. Is there anything special about your systems I should know about?

**Comment by majk-p** [comment]
Nothing special. quite fresh system install. Keybase installed from https://prerelease.keybase.io/keybase_amd64.deb. Just tried installing nightly and then downgrading - with no luck. 
Any cache/temp files we can remove maybe?

**Comment by carroarmato0** [comment]
Same here, nothing special. 
Made an strace for fun if anyone can spot anything of interest.
[keybase-strace.txt](https://github.com/keybase/client/files/4323264/keybase-strace.txt)

I agree with @majk-p, I think it's probably some weird state after the upgrade, clearing some kind of cache might maybe solve this?

**Comment by Currrupted** [comment]
> I updated Keybase through my package manager on Ubuntu 19.10, but now the UI seems to be crashing immediately when starting up.
> 
> I do have a crash report from the crash detection system on Ubuntu
> [_opt_keybase_Keybase.1000.crash.txt](https://github.com/keybase/client/files/4316998/_opt_keybase_Keybase.1000.crash.txt)

I have the same issues. The application will not open and simply crashes to the background process.

**Comment by heronhaye** [comment]
@majk-p @carroarmato0 shot in the dark, what glibc version do you have installed?

**Comment by carroarmato0** [comment]
> @majk-p @carroarmato0 shot in the dark, what glibc version do you have installed?

> ➜  ~ ldd --version             
> ldd (Ubuntu GLIBC 2.30-0ubuntu2.1) 2.30

Good old default 2.30 shipped with Ubuntu

**Comment by heronhaye** [comment]
@carroarmato0 Are you a developer? Would you want to try building the Keybase GUI to help us debug it? Unfortunately I still cannot reproduce the issue.

If you do, you can clone https://github.com/keybase/client/, cd `shared`, and then run `yarn modules` and `yarn start`. Let me know if that crashes for you as well - it should connect to your keybase service if it's running.

Otherwise, I can't really do much with the coredump since it wasn't generated from a build with debug symbols. If you could reproduce it on a VM/vagrant box, that'd help me a lot in debugging.

**Comment by heronhaye** [comment]
Actually, another thing you could try is: download and install electron 8.0.2 with debug symbols from https://github.com/electron/electron/releases/download/v8.0.2/electron-v8.0.2-linux-x64-debug.zip, and run `/path/to/that/electron/binary /opt/keybase/resources/app/desktop/dist/node.bundle.js`. It'll fail for another reason but I think it'll get far enough to cause the segfault. This is what it looks like on my machine:

```
(electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
(electron) 'setAutoHideMenuBar function' is deprecated and will be removed. Please use 'autoHideMenuBar property' instead.
menubar:  TypeError: Error processing argument at index 0, conversion failure from /opt/keybase/resources/app/desktop/assets/IconTemplate.png
    at Menubar.<anonymous> (/opt/keybase/resources/app/desktop/dist/node.bundle.js:1:21177)
    at step (/opt/keybase/resources/app/desktop/dist/node.bundle.js:1:17186)
    at Object.next (/opt/keybase/resources/app/desktop/dist/node.bundle.js:1:17302)
    at /opt/keybase/resources/app/desktop/dist/node.bundle.js:1:16141
    at new Promise (<anonymous>)
    at __awaiter (/opt/keybase/resources/app/desktop/dist/node.bundle.js:1:15751)
    at Menubar.appReady (/opt/keybase/resources/app/desktop/dist/node.bundle.js:1:20682)
    at App.<anonymous> (/opt/keybase/resources/app/desktop/dist/node.bundle.js:1:17983)
    at App.emit (events.js:215:7)
^CQuit through before-quit
Quit the app
Exec (null) not available for platform: darwin != linux
Done with ctlstop
exiting app
```

**Comment by carroarmato0** [comment]
> Actually, another thing you could try is: download and install electron 8.0.2 with debug symbols from https://github.com/electron/electron/releases/download/v8.0.2/electron-v8.0.2-linux-x64-debug.zip, and run `/path/to/that/electron/binary /opt/keybase/resources/app`. If that crashes too, could you send another crash report?

This is weird, I redownloaded the electron release multiple times, but each time I tried to execute it I got a 
```
➜  debug ./electron.debug 
zsh: exec format error: ./electron.debug
➜  debug file electron.debug 
electron.debug: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter *empty*, for GNU/Linux 3.2.0, BuildID[sha1]=6e608e0ace0d5785b97fb1a5769af07b9dec5e7e, with debug_info, not stripped

```

**Comment by carroarmato0** [comment]
This looks interesting though in 8.1.0, will give that one a try

**Comment by carroarmato0** [comment]
@heronhaye I'm not sure if I'm doing something wrong, but maybe I'm supposed to be calling electron through nodejs instead of executing it directly?

**Comment by heronhaye** [comment]
@carroarmato0 sorry for the bad instructions, I figured out how to load the debug symbols myself. It looks like it has something to do with your systray. What DE/WM are you using, and does it have a system tray?

**Comment by DaveLMSP** [comment]
I'm seeing the same issue on two machines running Linux Mint 19.3 with Mate desktop v1.22.  They're running two different kernels which are copied below:
Machine1: 5.0.0-32-generic #34~18.04.2-Ubuntu SMP Thu Oct 10 10:36:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
Machine2: 4.15.0-88-lowlatency #88-Ubuntu SMP PREEMPT Tue Feb 11 20:51:55 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

When I run keybase the main GUI opens, but the systray icon/menu stopped working with the most recent update.

**Comment by kstudent** [comment]
Same here: 
Linux version 4.18.0-25-generic (buildd@lcy01-amd64-025) (gcc version 8.3.0 (Ubuntu 8.3.0-6ubuntu1~18.10.1)) #26-Ubuntu SMP Mon Jun 24 09:32:08 UTC 2019

**Comment by carroarmato0** [comment]
@heronhaye  default in Ubuntu with Gnome Shell.  I tried switching between xOrg and Wayland, but that doesn't change anything, so can be excluded.

**Comment by kg4zow** [comment]
If it helps, this version of the Keybase client works just fine under Tails 4.4.

**Comment by ricardobalk** [comment]
Possible solution was found in #23073. Just waiting for the release.

**Comment by heronhaye** [comment]
@carroarmato0 @majk-p @Currrupted @kstudent @kg4zow please try out the latest nightly at https://keybase.io/docs/linux-user-guide#nightly-builds and let me know if it fixes the issue for you. I'll update when the stable release is out. Thanks!

@DaveLMSP if the GUI opens fine but it's just the systray icon that's broken, that's known and due to a breaking change in upstream Chromium.

**Comment by carroarmato0** [comment]
@heronhaye it's working!

**Comment by Currrupted** [comment]
> @carroarmato0 @majk-p @Currrupted @kstudent @kg4zow please try out the latest nightly at https://keybase.io/docs/linux-user-guide#nightly-builds and let me know if it fixes the issue for you. I'll update when the stable release is out. Thanks!
> 
> @DaveLMSP if the GUI opens fine but it's just the systray icon that's broken, that's known and due to a breaking change in upstream Chromium.

Wow, thanks a lot! I was able to open Keybase again and so far I haven't run into any issues! :heart:

**Comment by michal-pawlik** [comment]
> 
> 
> @carroarmato0 @majk-p @Currrupted @kstudent @kg4zow please try out the latest nightly at https://keybase.io/docs/linux-user-guide#nightly-builds and let me know if it fixes the issue for you. I'll update when the stable release is out. Thanks!
> 
> @DaveLMSP if the GUI opens fine but it's just the systray icon that's broken, that's known and due to a breaking change in upstream Chromium.

Thanks a lot, works for me as well!

**Comment by majk-p** [comment]
> 
> 
> @carroarmato0 @majk-p @Currrupted @kstudent @kg4zow please try out the latest nightly at https://keybase.io/docs/linux-user-guide#nightly-builds and let me know if it fixes the issue for you. I'll update when the stable release is out. Thanks!
> 
> @DaveLMSP if the GUI opens fine but it's just the systray icon that's broken, that's known and due to a breaking change in upstream Chromium.

Thank you, works great!

**Comment by heronhaye** [comment]
All right, we've released version 5.3.1 which should fix the issue. Please upgrade and let me know if it still doesn't work.

If you installed the nightly, you can downgrade to the release version (manually reinstall from https://keybase.io/docs/the_app/install_linux), or you can just stay on the nightly (you'll get an update for the next stable release when it comes out).
